great_expectations/cli/suite.py

```diff
@@ -323,6 +430,96 @@ def suite_edit(
+        interactive_flag: --interactive from the `suite new` CLI command
+        no_interactive_flag: --non-interactive from the `suite new` CLI command
+        datasource_name: --datasource-name from the `suite new` CLI command
+        batch_request: --batch-request from the `suite new` CLI command
+
+    Returns:
+        boolean of whether to enter interactive mode
+    """
+
+    error_message: Optional[str] = None
+
+    # Convert interactive / no-interactive flags to interactive
+    interactive: Optional[bool] = None
+    if interactive_flag is True and no_interactive_flag is True:
+        error_message = """Please choose either --interactive or --non-interactive, you may not choose both."""
+    elif interactive_flag is False and no_interactive_flag is False:
+        interactive = None
+    elif interactive_flag is True and no_interactive_flag is False:
+        interactive = True
+    elif interactive_flag is False and no_interactive_flag is True:
+        interactive = False
+
+    if (datasource_name is not None) and (batch_request is not None):
+        error_message = """Only one of --datasource-name DATASOURCE_NAME and --batch-request <path to JSON file> \
+options can be used.
+"""
+
+    if error_message is not None:
+        cli_message(string=f"<red>{error_message}</red>")
+        toolkit.send_usage_message(
+            data_context=context, event=usage_event_end, success=False
+        )
+        sys.exit(1)
+
+    # If user has provided a flag determining their configuration, skip prompt.
+    if (
+        (interactive is not None)
+        or (datasource_name is not None)
+        or (batch_request is not None)
+    ):
+        if (datasource_name is not None) and (
+            interactive is False or interactive is None
+        ):
+            cli_message(
+                "<green>Entering interactive mode since you passed the --datasource-name flag</green>"
+            )
+            interactive = True
+        elif (batch_request is not None) and (
+            interactive is False or interactive is None
+        ):
+            cli_message(
+                "<green>Entering interactive mode since you passed the --batch-request flag</green>"
+            )
+            interactive = True
+    else:
+        suite_edit_method = click.prompt(
+            """\
+How would you like to edit your Expectation Suite?
+    1. Manually, without interacting with a sample batch of data (default)
+    2. Interactively, with a sample batch of data
+""",
+            type=click.Choice(["1", "2"]),
+            show_choices=False,
+            default="1",
+            show_default=False,
+        )
+        # Default option
+        if suite_edit_method == "":
+            interactive = False
+        if suite_edit_method == "1":
+            interactive = False
+        elif suite_edit_method == "2":
+            interactive = True
+
+    return interactive
+
+
```

Comment:
@anthonyburdi Is it possible to unify the `_process_suite_edit_flags_and_prompt` and `_process_suite_new_flags_and_prompt` within a single method?..

Reply:
I thought about this but I think it's that uncanny valley you spoke about. There is a bit too much changed between the two (both functionality and test cases) for a simple merger. I think it is potentially a good refactoring opportunity for the future, if not merging then at least pulling out some of the duplicated code.